operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -307,30 +307,29 @@ func (c *cesMgr) InsertCEPInCache(cep *cilium_v2.CoreCiliumEndpoint, ns string)
 
 	// If given cep object isn't packed in any of the CES. find a new ces
 	// to pack this cep.
-	cb := func() *cesTracker {
+	ces := func() string {
 		// get first available CES
 		for _, ces := range c.desiredCESs.getAllCESs() {
-			ces.backendMutex.RLock()
+			ces.mutex.RLock()
+			defer ces.mutex.RUnlock()
```

Comment:
I don't think that this defer will work as you intend in the `continue` case below; defer runs at the end of the overall function, not the current scope.

Reply:
^^ I guess technically with an RLock this is OK, we can grab it as many times as we like, as long as we balance the unlocks. But I'm not sure if that was intentional; equivalent logic could instead be implemented by just moving the locking out of the loop, and then we wouldn't lock for each iteration.

Reply:
Yea, you are right, I did not think this through, will do refactor in another PR and focus this one on fixing the issue.